src/Maker/MakeEntity.php

```diff
@@ -80,4 +171,117 @@ public function configureDependencies(DependencyBuilder $dependencies)
+            }
+            // check reserved words
+            if ($this->isReservedKeyword($name)) {
+                throw new \InvalidArgumentException(sprintf('Name "%s" is a reserved word.', $name));
+            }
+            // check for valid PHP variable name
+            if (!is_null($name) && !Str::isValidPhpVariableName($name)) {
+                throw new \InvalidArgumentException(sprintf('"%s" is not a valid PHP variable name.', $name));
+            }
+
+            return $name;
+        });
+
+        if (!$fieldName) {
+            return;
+        }
+
+        $defaultType = 'string';
+        // try to guess the type by the field name prefix/suffix
+        // convert to camel case for simplicity
+        $camelCasedField = Str::snakeCase($fieldName);
+        if (substr($camelCasedField, -3) == '_at') {
+            $defaultType = 'datetime';
+        } elseif (substr($camelCasedField, -3) == '_id') {
+            $defaultType = 'integer';
+        } elseif (substr($camelCasedField, 0, 3) == 'is_') {
+            $defaultType = 'boolean';
+        } elseif (substr($camelCasedField, 0, 4) == 'has_') {
+            $defaultType = 'boolean';
+        }
+
+        $type = null;
+        while (null === $type) {
+            $question = new Question('Field type (enter <comment>?</comment> to see all types)', $defaultType);
+            $question->setAutocompleterValues(Type::getTypesMap());
+            $type = $io->askQuestion($question);
+
+            if ('?' === $type) {
+                $this->printAvailableTypes($io);
+                $io->writeln('');
+
+                $type = null;
+            } elseif (!array_key_exists($type, Type::getTypesMap())) {
+                $io->error(sprintf('Invalid type "%s".', $type));
+                $this->printAvailableTypes($io);
+                $io->writeln('');
+
+                $type = null;
+            }
+        }
+
+        $data = array('fieldName' => $fieldName, 'type' => $type);
+        if ($type == 'string') {
+            // default to 255, avoid the question
+            $data['length'] = 255;
+        } elseif ('decimal' === $type) {
+            // 10 is the default value given in \Doctrine\DBAL\Schema\Column::$_precision
+            $data['precision'] = $io->ask('Precision', 10, [Validator::class, 'validatePrecision']);
+
+            // 0 is the default value given in \Doctrine\DBAL\Schema\Column::$_scale
+            $data['precision'] = $io->ask('Scale', 0, [Validator::class, 'validateScale']);
+        }
+
+        $question = new Question('Can field be null (nullable)?', 'false');
+        $question->setValidator([Validator::class, 'validateBoolean']);
+        $question->setAutocompleterValues(['true', 'false']);
+        if ($nullable = $io->askQuestion($question)) {
+            $data['nullable'] = $nullable;
+        }
+
+        return $data;
+    }
+
+    private function printAvailableTypes(ConsoleStyle $io)
```

Comment:
We could simplify this method as follows:

```php
private function printAvailableTypes(ConsoleStyle $io)
{
    $io->write(' <info>Available types:</info> ');

    $types = array_keys(Type::getTypesMap());
    $io->write(wordwrap(implode(', ', $types), 50));
    $io->writeln('');
}
```

And if you want to keep the `<comment>` styling (adjust the word-wrap line-length accordingly):

```php
private function printAvailableTypes(ConsoleStyle $io)
{
    $io->write(' <info>Available types:</info> ');

    $types = array_map(function ($v) { return '<comment>'.$v.'</comment>'; }, array_keys(Type::getTypesMap()));
    $io->write(wordwrap(implode(', ', $types), 80));
    $io->writeln('');
}
```

Reply:
Actually, I just rewrote this algorithm to be more complex, but make the output much fancier and easier for the user.